src/decode/lzma2.rs

```diff
@@ -1,27 +1,24 @@
 use crate::decode::lzbuffer;
 use crate::decode::lzbuffer::LzBuffer;
 use crate::decode::lzma;
+use crate::decode::lzma::DecoderState;
 use crate::decode::lzma::LzmaProperties;
 use crate::decode::rangecoder;
 use crate::error;
 use byteorder::{BigEndian, ReadBytesExt};
 use std::io;
 use std::io::Read;
 
-pub fn decode_stream<R, W>(input: &mut R, output: &mut W) -> error::Result<()>
+fn decode_stream<R, W>(
```

Comment:
Nit: it might be worth moving these free functions into the `impl` block of the new `Lzma2Decoder` struct?

Reply:
The only thing is the diff is a little more hard to read now but that doesn't really matter anymore.

Reply:
I moved `lzma2::decode_stream` directly into `Lzma2Decoder::decompress` and just made it take `&mut self`.